### PR TITLE
Tidy up validate-modules:doc-required-mismatch

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Tidy up on sanity tests' ignore files, removing lines with validation-modules for these modules.

The focus was to pass the tests so no code was changed other than adjusting argument_spec for missing types or inconsistencies with the docs. The priority has been not to change any functionality, but to ensure that code and docs are synchronized.

Some lines were added back to the files, but a total of 129 lines were removed (among 2.9, 2.10 and 2.11).

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
plugins/modules/storage/zfs/zfs_delegate_admin.py
plugins/modules/storage/ibm/ibm_sa_vol_map.py
plugins/modules/storage/ibm/ibm_sa_vol.py
plugins/modules/storage/ibm/ibm_sa_pool.py
plugins/modules/storage/ibm/ibm_sa_host_ports.py
plugins/modules/storage/ibm/ibm_sa_host.py
plugins/modules/storage/ibm/ibm_sa_domain.py
plugins/modules/system/aix_devices.py
plugins/modules/system/osx_defaults.py
plugins/modules/storage/glusterfs/gluster_peer.py
plugins/modules/source_control/hg.py
plugins/modules/source_control/gitlab/gitlab_runner.py
plugins/modules/source_control/gitlab/gitlab_hook.py
plugins/modules/source_control/gitlab/gitlab_deploy_key.py
plugins/doc_fragments/nios.py
plugins/modules/remote_management/oneview/oneview_fc_network.py
plugins/modules/remote_management/manageiq/manageiq_provider.py
plugins/modules/remote_management/cobbler/cobbler_system.py
plugins/modules/remote_management/cobbler/cobbler_sync.py
plugins/modules/net_tools/nios/nios_a_record.py
plugins/modules/net_tools/nios/nios_aaaa_record.py
plugins/modules/net_tools/nios/nios_cname_record.py
plugins/modules/net_tools/nios/nios_dns_view.py
plugins/modules/net_tools/nios/nios_fixed_address.py
plugins/modules/net_tools/nios/nios_host_record.py
plugins/modules/net_tools/nios/nios_member.py
plugins/modules/net_tools/nios/nios_mx_record.py
plugins/modules/net_tools/nios/nios_naptr_record.py
plugins/modules/net_tools/nios/nios_ptr_record.py
plugins/modules/net_tools/nios/nios_srv_record.py
plugins/modules/net_tools/nios/nios_txt_record.py
plugins/modules/net_tools/nios/nios_zone.py
plugins/modules/net_tools/ldap/ldap_passwd.py
plugins/modules/net_tools/dnsmadeeasy.py
plugins/modules/files/xml.py
plugins/modules/database/vertica/vertica_configuration.py
plugins/modules/database/mssql/mssql_db.py
plugins/modules/database/misc/redis.py
plugins/modules/clustering/pacemaker_cluster.py
plugins/modules/clustering/consul/consul_kv.py
plugins/modules/clustering/consul/consul_acl.py
plugins/modules/cloud/spotinst/spotinst_aws_elastigroup.py
plugins/modules/cloud/smartos/vmadm.py
plugins/modules/cloud/packet/packet_volume_attachment.py
plugins/modules/cloud/packet/packet_ip_subnet.py
plugins/modules/cloud/ovirt/ovirt_scheduling_policy_facts.py
plugins/doc_fragments/online.py

### CODE REVIEW

File: plugins/modules/cloud/smartos/vmadm.py [hunk 1]
```diff
@@ -604,7 +659,7 @@ def main():
             'zfs_root_compression', 'zpool'
         ],
         'bool': [
-            'archive_on_delete', 'autoboot', 'debug', 'delegate_dataset',
```
Comment (felixfontein) [inline plugins/modules/cloud/smartos/vmadm.py:607]: This change breaks backwards compatibility. The option needs to be deprecated first.

Could you please move that part to another PR? This is the only thing that cannot be backported to stable-1 (which from now on only accepts bugfixes and no longer new deprecations).

File: plugins/modules/cloud/smartos/vmadm.py [hunk 2]
```diff
@@ -501,13 +560,9 @@ def create_payload(module, uuid):
...
-    vmattrs = filter(lambda prop: prop not in module_options, p)
-
-    vmdef = {}
-
-    for attr in vmattrs:
-        if p[attr]:
-            vmdef[attr] = p[attr]
+    # @TODO make this a simple {} comprehension as soon as py2 is ditched
+    # @TODO {k: v for k, v in p.items() if k not in module_options}
+    vmdef = dict([(k, v) for k, v in p.items() if k not in module_options])
```
Comment (felixfontein) [inline plugins/modules/cloud/smartos/vmadm.py:565]:
```suggestion
    vmdef = dict([(k, v) for k, v in p.items() if k not in module_options and v])
```

File: changelogs/fragments/1415-valmod_req_mismatch.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - vmadm - removed unused ``debug`` parameter (https://github.com/ansible-collections/community.general/pull/1415).
```
Comment (felixfontein) [inline changelogs/fragments/1415-valmod_req_mismatch.yml:2]:
```suggestion
  - vmadm - simplification of code (https://github.com/ansible-collections/community.general/pull/1415).
```

Comment (russoz) [reply changelogs/fragments/1415-valmod_req_mismatch.yml]: oopsie, I was faster.. lemme get the file back

File: plugins/modules/cloud/smartos/vmadm.py [hunk 4]
```diff
@@ -501,13 +560,9 @@ def create_payload(module, uuid):
...
-    vmattrs = filter(lambda prop: prop not in module_options, p)
-
-    vmdef = {}
-
-    for attr in vmattrs:
-        if p[attr]:
-            vmdef[attr] = p[attr]
+    # @TODO make this a simple {} comprehension as soon as py2 is ditched
+    # @TODO {k: v for k, v in p.items() if k not in module_options}
+    vmdef = dict([(k, v) for k, v in p.items() if k not in module_options and v])
```
Comment (felixfontein) [inline plugins/modules/cloud/smartos/vmadm.py:565]: I think for sake of readability, we should get rid of the `p` variable, since right now it is only used in precisely one place:
```suggestion
    vmdef = dict([(k, v) for k, v in mdule.params.items() if k not in module_options and v])
```
(Also remove its definition above)

Comment (russoz) [reply plugins/modules/cloud/smartos/vmadm.py]: done, and I'm signing off for today. see you on the other side

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for fixing more and more :)

**Comment by felixfontein** [comment]
Backport in #1417.

**Comment by russoz** [comment]
Always a pleasure, thanks for reviewing!
